src/katsdpcontroller/generator.py

```diff
@@ -208,6 +208,63 @@ def aggregate_compute(self) -> Tuple[Sensor.Status, int]:
         return (status, self._total)
 
 
+class SyncSensor(SimpleAggregateSensor[bool]):
+    """Aggregate which takes the logical AND of its children.
+
+    The status computes as ERROR if any of the child readings are False, and
+    FAILURE if they aren't all present.
+    """
+
+    def __init__(
+        self, target: SensorSet, sensor_type: Type[bool],
+        name: str, description: str, units: str = "",
+        *,
+        auto_strategy: Optional["SensorSampler.Strategy"] = None,
+        auto_strategy_parameters: Iterable[Any] = (),
+        name_regex: re.Pattern,
+        children: int
+    ) -> None:
+        self.target = target
+        self.name_regex = name_regex
+        self.children = children
+        self._known = 0
+        self.synchronised = False
+        self.curr_values: Dict[str, bool] = {}
+
+        super().__init__(
+            target, sensor_type, name, description, units,
+            auto_strategy=auto_strategy,
+            auto_strategy_parameters=auto_strategy_parameters
+        )
+
+    def filter_aggregate(self, sensor: Sensor) -> bool:
+        return bool(self.name_regex.fullmatch(sensor.name))
+
+    def aggregate_add(self, sensor: Sensor[_T], reading: Reading[_T]) -> bool:
+        assert isinstance(reading.value, bool)
+        if reading.status.valid_value():
+            self.curr_values[sensor.name] = reading.value
```

Comment:
Not sure why you need to keep a dictionary by sensor. All you need is a count of how many are synchronised, which `aggregate_compute` will compare to `children`.